### DISCUSSION TITLE
How to use plugins programmatically?

### DISCUSSION THREAD

**Discussion by JWebDev** [GoogleChrome/lighthouse#15119] [Q&A]
Hi all, 

I'm running lighthouse programmatically. Everything is fine, but if I add `lighthouse-plugin-publisher-ads` to my config, then I expect it in generated json categories section.

My lighthouse config looks like. 
After that, i should see `lighthouse-plugin-publisher-ads` in `categories` object. If I run lighthouse from command line, then generated json has `lighthouse-plugin-publisher-ads` in `categories` section. If I run it programmatically, then `lighthouse-plugin-publisher-ads` does not exists in generated json report in `categories` object

`runnerResult = await lighthouse(url, flags, configs, page)`
configs object
```
{
   "extends": "lighthouse:default",
   "settings": {
      "formFactor": "desktop",
      "disableFullPageScreenshot": false,
      "locale": "en",
      "screenEmulation": {
         "disabled": true
      },
      "onlyCategories": [
         "performance",
         "accessibility",
         "best-practices",
         "seo",
         "pwa"
      ]
   },
   "plugins": [
      "lighthouse-plugin-publisher-ads"
   ]
}
```
`lighthouse-plugin-publisher-ads` is in package.json
```
  "dependencies": {
    "lighthouse": "^10.2.0",
    "lighthouse-plugin-publisher-ads": "1.5.7-beta",
    "puppeteer": "^19.9.1"
  },
  "overrides": {
    "lighthouse-plugin-publisher-ads": {
      "lighthouse": "$lighthouse"
    }
  }
```

Thanks for the help.

**Comment by adamraine** [comment]
https://github.com/GoogleChrome/lighthouse/issues/15140
